src/lib/core/popen.c

```diff
@@ -1178,7 +1178,13 @@ popen_new(struct popen_opts *opts)
 	 * vfork to complete. Also we try to do as minimum
 	 * operations before the exec() as possible.
 	 */
+#pragma GCC diagnostic push
```

Comment:
What about clang compiler?

Reply:
@locker clang's diagnostic pragmas are [backwards compatible](https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas) with those of GCC .